collectd/files/plugin/collectd_glusterfs.py

```diff
@@ -104,103 +106,112 @@ def itermetrics(self):
             }
 
         # Collect volumes' metrics
-        out, err = self.execute(
-            [GLUSTER_BINARY, 'volume', 'status', 'all', 'detail'],
-            shell=False)
+        cmd = [GLUSTER_BINARY, 'volume', 'status', 'all', 'detail']
+        out, err = self.execute(cmd, shell=False, log_error=False)
         if not out:
-            raise base.CheckException("Failed to execute 'gluster volume'")
-
-        for vol_block in vol_status_re.split(out):
-            volume_m = volume_re.search(vol_block)
-            if not volume_m:
-                continue
-            volume = volume_m.group('volume')
-            for line in vol_block_re.split(vol_block):
-                peer_m = brick_server_re.search(line)
-                if not peer_m:
+            if err and vol_status_transaction_in_progress_re.match(err):
+                # "transaction already in progress" error, we assume volumes
+                # metrics are being collected on another glusterfs node, and
+                # just silently skip the collecting of the volume metrics
```

Comment:
It's the same, but I think "collecting" is more about the act of gathering… But I am not a native-english speaker :)